power_grid_model_io_native_c/power_grid_model_io_native_c/include/power_grid_model_io_native_c/vnf_pgm_converter.h

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#include "../../../../../../power-grid-model/power_grid_model_c/power_grid_model_c/src/forward_declarations.hpp"
+
+#pragma once
+#ifndef POWER_GRID_MODEL_IO_NATIVE_C_VNF_PGM_CONVERTER_H
+#define POWER_GRID_MODEL_IO_NATIVE_C_VNF_PGM_CONVERTER_H
+
+#ifdef __cplusplus
+
+class PgmVnfConverter;
+
+extern "C" {
+#endif
+
+PgmVnfConverter* parse_vnf_file_wrapper(PgmVnfConverter* obj);
+
+PGM_ConstDataset* convert_input_wrapper(PgmVnfConverter* obj);
+
+#ifdef __cplusplus
+}
+#endif
+
+
+class PgmVnfConverter {
+    public:
+        // Public attributes
+        char* f_file_buffer;
+        PGM_WritableDataset* deserialized_data; // this type because it is generated by a deserializer type structure
```

Comment:
public member values are bad practice if you also have private ones. please have the constructor set them and/or add setters and/or getters